lib/services/nodes-api-service.js

```diff
@@ -280,6 +280,11 @@ function nodeApiServiceFactory(
                     }
                 };
                 return workflowApiService.createAndRunGraph(configuration);
+            } else if (node.type === Constants.NodeTypes.Rack) {
+                return eventsProtocol.publishNodeEvent(node, 'added')
```

Comment:
I don't know if the event should be triggered for other types of nodes since there's already a 'discovered' event which should be triggered for most node types. "rack" nodes don't have a discovery process though hence this "added" event. 

I think that createNodeAndRunDiscovery() is a method of profiles-api-service just for internal use?